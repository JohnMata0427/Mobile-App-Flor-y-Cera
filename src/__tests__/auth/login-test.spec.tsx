import LoginScreen from '@/app/(auth)/login';
import { render, waitFor } from '@testing-library/react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';

const wrapAllProviders = (children: React.ReactNode) => {
  return <SafeAreaProvider>{children}</SafeAreaProvider>;
};

describe('Pantalla de autenticación', () => {
  let screen: ReturnType<typeof render>;

  beforeAll(() => {
    screen = render(<LoginScreen />, {
      wrapper: wrapAllProviders,
    });
  });

  test('Debería renderizar la pantalla de inicio de sesión', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('Debería iniciar sesión con las credenciales', async () => {
    const { findByTestId } = screen;

    const emailInput = await findByTestId('input-field-email');
    const passwordInput = await findByTestId('input-field-password');

    const loginButton = await findByTestId('button-login');

    await waitFor(() => {
      emailInput.props.onChangeText('test@example.com');
      passwordInput.props.onChangeText('password');
      loginButton.props.onPress();
    });
  });
});
