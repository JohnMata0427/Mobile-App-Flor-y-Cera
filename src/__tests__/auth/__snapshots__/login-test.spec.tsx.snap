// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pantalla de autenticación Debería renderizar la pantalla de inicio de sesión 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
